### PR TITLE
Simplify HttpContextAccessor (now singleton)

### PR DESCRIPTION
**Pull request by HaoK**
MVC tests all pass with this change so looks good I think

cc @davidfowl @lodejard @divega

### CODE REVIEW

**Comment by HaoK** [comment]
@davidfowl Restored the callcontext/asynclocal logic

File: src/Microsoft.AspNet.Hosting/HttpContextAccessor.cs [hunk 1]
```diff
@@ -14,60 +14,39 @@ namespace Microsoft.AspNet.Hosting
...
-            CallContext.LogicalSetData(LogicalDataKey, new ObjectHandle(httpContext));
-            return prior != null ? prior.Unwrap() as HttpContext : null;
-#elif ASPNETCORE50
-            var prior = _httpContextCurrent.Value;
-            _httpContextCurrent.Value = httpContext;
-            return prior;
 #else
-            return null;
+        // REVIEW: or should this throw?
+        public HttpContext Value { get; set; }
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Hosting/HttpContextAccessor.cs]: What is this? Nuke it, there is no else

File: test/Microsoft.AspNet.TestHost.Tests/TestServerTests.cs [hunk 2]
```diff
@@ -38,6 +38,23 @@ public void ThrowsIfNoApplicationEnvironmentIsRegisteredWithTheProvider()
...
 
         [Fact]
+        public async Task CanAccessHttpContext()
+        {
+            var services = new ServiceCollection().BuildServiceProvider();
+            TestServer server = TestServer.Create(app =>
+            {
+                app.Run(context =>
+                {
+                    var accessor = app.ApplicationServices.GetRequiredService<IHttpContextAccessor>();
```
Comment (davidfowl) [inline test/Microsoft.AspNet.TestHost.Tests/TestServerTests.cs]: A better test would to be to do this to a singleton service and called from here

**Comment by davidfowl** [comment]
After the changes, :shipit: !

**Comment by HaoK** [comment]
Merged 16a811479eb4f098387b0bcd6702e25003ae6c59 still need to update the test

**Pull request closed without merging**